firebird/driver.py

```diff
@@ -0,0 +1,178 @@
+import firebird, ccd, argparse, json, hashlib
+from firebird import products
+from pandas import to_datetime
+from firebird  import SPARK_MASTER, SPARK_EXECUTOR_IMAGE, SPARK_EXECUTOR_CORES, SPARK_EXECUTOR_FORCE_PULL
+from firebird  import AARDVARK_SPECS_URL
+from firebird  import simplify_objects, dtstr_to_ordinal
+from .validation import *
+from .aardvark import pyccd_tile_spec_queries, chip_specs
+from .chip     import ids as chip_ids
+from pyspark   import SparkConf, SparkContext
+from datetime  import datetime
+from .cassandra import insert_statement
+from .cassandra import execute as cassandra_execute
+from .cassandra import RESULT_INPUT
+
+# to be replaced by dave's official function names
+#
+from .aardvark import spicey_meatball, spicey_meatball_dates
+#
+#
+
+parser = argparse.ArgumentParser(description="Driver for LCMAP product generation")
+# could get cute here and loop this, leaving unwound for now
+parser.add_argument('-ord', dest='ord_date')
+parser.add_argument('-acq', dest='acquired')
+parser.add_argument('-ulx', dest='ulx')
+parser.add_argument('-uly', dest='uly')
+parser.add_argument('-lrx', dest='lrx')
+parser.add_argument('-lry', dest='lry')
+parser.add_argument('-lastchange', dest='lastchange', action='store_true')
+parser.add_argument('-changemag',  dest='changemag',  action='store_true')
+parser.add_argument('-changedate', dest='changedate', action='store_true')
+parser.add_argument('-seglength',  dest='seglength',  action='store_true')
+parser.add_argument('-qa',         dest='qa',         action='store_true')
+
+args = parser.parse_args()
+
+
+def simplify_detect_results(results):
+    ''' Convert child objects inside CCD results from NamedTuples to Dictionaries '''
+    output = dict()
+    for key in results.keys():
+        output[key] = simplify_objects(results[key])
+    return output
+
+
+def detect(input, tile_x, tile_y):
```

Comment:
you're referring to the 'input' object?   its not necessary, more just an evolution from how we started using spark initially, and the functions used to get the data organized for ccd